java/src/server/games/ServerManager.java

```diff
@@ -80,10 +82,13 @@ public void stopServer() {
      * returns that to be the servers current database
      *
      * @param plugins
+     * @param name
      * @return PersistenceProvider for server
      */
     @Override
-    public IPersistenceProvider getPersistenceProvider(List<IPlugin> plugins) {
-        return null;
+    public IPersistenceProvider getPersistenceProvider(List<AbstractPlugin> plugins, String name) {
+        return (IPersistenceProvider) plugins.stream()
+                .filter(x -> x.getType() == PERSISTENCE && name.equals(x.getName()))
+                .collect(Collectors.toList()).get(0);
```

Comment:
Use `findFirst().orElse(null);`. There's no reason to collect into a list if you're only getting the first element.